locksmith/src/utils/claim.ts

```diff
@@ -0,0 +1,36 @@
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import { networks } from '@unlock-protocol/networks'
+import { ethers } from 'ethers'
+
+interface Options {
+  recipients: string[]
+  data?: (string | null)[]
+  lockAddress: string
+  network: number
+}
+
+export const getTotalPurchasePriceInCrypto = async ({
+  recipients,
+  network,
+  data,
+  lockAddress,
+}: Options) => {
+  const web3Service = new Web3Service(networks)
+  const prices = await Promise.all(
+    recipients.map((recipient, index) => {
+      return web3Service.purchasePriceFor({
+        lockAddress,
+        userAddress: recipient,
+        network,
+        data: data?.[index] || '0x',
+        referrer: recipient,
```

Comment:
In the future we may wat to use the *real* referrer, because in theory the lock could change its price based on it... but let's not worry too much about it right now

Reply:
I actually have the code for this. I added it previously but then removed it because when we do any TX ourselves, we use our purchaser address as referrer.